biorxiv/article_distances/biorxiv_corrected_published_rate.py

```diff
@@ -0,0 +1,520 @@
+print(len(missing_link_mapper))
+
+
+# In[10]:
+
+
+updated_biorxiv_journal_df = (
+    pd.DataFrame
+    .from_records([
+        {
+            "doi":row[0], 
+            "published":row[1][0], 
+            "pmcoa":row[1][1]
+        } 
+        for row in list(remaining_links.items())
+    ])
+    .merge(
+        biorxiv_journal_df
+         .drop(["published", "pmcoa"], axis=1), 
+        on="doi"
+    )
+)
+print(updated_biorxiv_journal_df.shape)
+updated_biorxiv_journal_df.head()
+
+
+# ## Gather date published for each document
+
+# In[11]:
+
+
+file_path = Path("output/biorxiv_articles_published_date.tsv")
+
+
+# In[12]:
+
+
+if not file_path.exists():
+    data_records = []
+    xml_parser = ET.XMLParser(encoding='UTF-8', recover=True)
+    for document in tqdm(updated_biorxiv_journal_df.document.tolist()):
+        root = (
+            ET
+            .parse(
+                f"../biorxiv_articles/{document}", 
+                parser=xml_parser
+            )
+            .getroot()
+        )
+
+        date_node = root.xpath(
+            "//front/article-meta/history/date"
+        )
+
+        if len(date_node) == 0:
+            print("Error in xml:")
+            print(document)
+            continue
+
+        elif len(date_node) < 2:
+            date_node = date_node[0]
+
+        else:
+            date_node = date_node[1]
+
+        data_row = dict(
+            map(
+                lambda x: (x.tag, x.text), 
+                date_node.getchildren()
+            )
+        )
+
+        data_records.append({
+            "month":f"{data_row['year']}-{int(data_row['month']):02d}",
+            "document":document
+        })
+
+
+# In[13]:
+
+
+if not file_path.exists():
+    published_preprint_df = pd.DataFrame.from_records(
+        data_records+[
+            {"month":"2018-09", "document":"423517_v1.xml"},
+            {"month":"2019-01", "document":"528299_v1.xml"}
+        ]
+    )
+else:
+    published_preprint_df = pd.read_csv(
+        file_path, 
+        sep="\t"
+    )
+print(published_preprint_df.shape)
+published_preprint_df.head()
+
+
+# In[14]:
+
+
+if not file_path.exists():
+    final_mapped_df = (
+        published_preprint_df
+        .merge(
+            updated_biorxiv_journal_df, 
+            on="document"
+        )
+        .rename(index=str, columns={"month":"pub_month"})
+    )
+    print(final_mapped_df.shape)
+else:
+    final_mapped_df = published_preprint_df
+    print(final_mapped_df.shape)
+
+final_mapped_df
+
+
+# In[15]:
+
+
+if not file_path.exists():
+    (
+        final_mapped_df
+        .to_csv(
+            "output/biorxiv_articles_published_date.tsv",
+            sep="\t", index=False
+        )
+    )
+
+
+# In[16]:
+
+
+snapshot_wo_links_df = (
+    biorxiv_journal_df
+    .assign(published_closed=lambda x: (x.pmcoa != x.published))
+    .merge(
+        final_mapped_df[["document", "pub_month"]],
+        on="document"
+    )
+    .replace({"2013-10":"2013-11"})
```

Comment:
> Why are you replacing this here? Seems like a one-off thing.

I replaced these values to be in line with Abdill et al's results. They start at 2013-11 instead of 2013-10.